sublime_music/ui/player_controls.py

```diff
@@ -805,6 +821,7 @@ def filename_to_pixbuf(
         self.volume_slider.connect("value-changed", self.on_volume_change)
         box.pack_start(self.volume_slider, True, True, 0)
 
+        vbox.pack_start(self.rating_buttons_box, False, True, 0)
```

Comment:
the rating box should only be added if the current adapter is can set ratings

Reply:
I have rework this a bit. There are now two settings for the adapter: whether the adapter can get the ratings or can set it. I have tested the different possibilities:

- An adapter that cannot get the ratings won't show any rating widget.
- An adapter that can get the ratings but cannot set it will show the rating widget but clicking on the stars to change it does not have any effect, either on the GUI or the server, obviously.
- For an adapter that supports both getting and setting the ratings the behaviour is as the original patch.